### ISSUE TITLE
[tags plugin] Add tags icons on tags index

### ISSUE THREAD

**Issue by Guts** [squidfunk/mkdocs-material#4100]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... the documentation does not mention anything about my idea
- [X] ... there are no open or closed issues that are related to my idea

### Description

Tag icons are great but they don't appear on the tags index page. It would be nice to have consistent icons between page, search and tags index.

I've tested with a quick setup:

`mkdocs.yml`

```yml
site_name: My Docs

plugins:
  - tags:
      tags_file: tags.md



theme:
  name: material
  icon:
    tag:
      html: fontawesome/brands/html5
      js: fontawesome/brands/js
      css:  fontawesome/brands/css3


extra:
  tags:
    test: html
```

`docs/index.md` :

```markdown
---
tags:
    - test
    - trigger
---

# Test

tags icons tester
```

`docs/tags.md`:

```markdown
# Tags

Following is a list of relevant tags:

[TAGS]
```

### Use Cases

Every website using tag icons.

### Screenshots / Mockups

Icon tag is present on page:

![image](https://user-images.githubusercontent.com/1596222/177786165-effed4ca-84ec-4658-b5e1-e67d0909bbe3.png)

On search:

![image](https://user-images.githubusercontent.com/1596222/177786502-37abf59d-3074-43a1-a0c5-2f85f3b143ff.png)

But not on index:

![image](https://user-images.githubusercontent.com/1596222/177786223-4182b837-a836-4e7d-bbd7-04a70e0593bd.png)

**Comment by squidfunk** [comment]
Fixed in squidfunk/mkdocs-material-insiders@a733eca. Tag icons were not forgotten on the tags index page, but the mapping was missing – the tag name was used, not the icon mapping. Furthermore, search will now also display the `#` symbol, when no specific tag was set, but icons are used.

**Comment by Guts** [comment]
Thanks! So it was a mini minor bug ;)

> furthermore, search will now also display the `#` symbol, when no specific tag was set, but icons are used.

To be sure, in the case where a default icon is defined it is this one that will be used?
